src/hera/task.py

```diff
@@ -124,6 +129,8 @@ class Task:
         Whether to continue task chain execution when this task fails.
     continue_on_error: bool = False
         Whether to continue task chain execution this task errors.
+    exit_hook: Optional['Task'] = None
```

Comment:
the `__init__` was kind of expected to become fairly rich. So far Hera avoided adding action-related parameters on the `__init__`. Rather, it favored adding methods on the `Task` that set a specific reaction on some result, such as `on_failure`. To avoid the proliferation of parameters that are not action related in the `__init__` I wonder what you think about adding an `on_exit(..., t: `Task`) -> 'Task'`? Similar to [on_success](https://github.com/argoproj-labs/hera-workflows/blob/main/src/hera/task.py#L302)!